v3/src/hooks/use-drop-hint.ts

```diff
@@ -0,0 +1,76 @@
+import {useDndContext} from "@dnd-kit/core";
+import {useDataSetContext} from "./use-data-set-context";
+import {AxisPlace, axisPlaceToAttrPlace} from"../components/graph/models/axis-model";
+import {useDataConfigurationContext} from "../components/graph/hooks/use-data-configuration-context";
+import {getStringWithSwaps} from "../utilities/tokened-string";
+
+export interface IUseDropHintProps {
+  place?: AxisPlace,
+  dropId?: string
+}
+
+function determineBaseString(placeString: string,  dropType: string, existingType: string){
+  const stringMap = {
+    bottom: {
+      numeric: {
+        empty: "addToEmptyPlace",
+        existing: "replaceAttribute"
+      },
+      categorical: {
+        empty: "layoutPlotsSideBySide",
+        existing: "layoutPlotsSideBySide"
+      }
+    },
+    left: {
+      numeric: {
+        empty: "addToEmptyPlace",
+        existing: "replaceAttribute"
+      },
+      categorical: {
+        empty: "layoutPlotsVertically",
+        existing: "layoutPlotsVertically"
+      }
+    },
+    // in V2 a drop on empty plot puts attr on X axis ("addToEmptyX")
+    // in v2 a drop on plot with a graph with attr on X axis results in color by attribute value
+    // below is based on existing v3 behavior - direct to color points by attribute value
+    top: {
+      numeric: {
+        empty: "dropInPlot",
+        existing: "dropInPlot"
+      },
+      categorical: {
+        empty: "dropInPlot",
+        existing: "dropInPlot"
+      }
+    }
+  }
+
+  const placeParam = placeString as keyof typeof stringMap,
+    typeParam = dropType as keyof typeof stringMap.bottom,
+    containsParam = existingType === undefined ? "empty" : "existing",
+    stringKey = stringMap[placeParam][typeParam][containsParam],
+    baseString = `DG.GraphView.${stringKey}`;
+  return baseString
+}
+
+export const useDropHint = ({place, dropId} : IUseDropHintProps) => {
+  const dataConfiguration = useDataConfigurationContext(),
+        dataSet = useDataSetContext(),
+        { active } = useDndContext()
+
+  if (active?.data.current && dataConfiguration && dataSet){
+    const placeString = place ? place : "top", // -> legend?
+      thisAxisPlace = axisPlaceToAttrPlace[placeString],
+      priorAttrId = dataConfiguration?.attributeID(thisAxisPlace),
+      priorAttrType = dataConfiguration?.attributeType(thisAxisPlace),
+      priorAttrName = dataSet?.attrFromID(priorAttrId as string)?.name,
+      attrId = active.data.current.attributeId,
+      attrName = dataSet.attrFromID(attrId).name,
+      attrType = dataSet.attrFromID(attrId).type as string,
+      details = priorAttrId ? [priorAttrName, attrName] : [attrName],
+      baseString = determineBaseString(placeString, attrType, priorAttrType),
+      hintText = getStringWithSwaps(baseString, "%@", details);
+    return hintText
+  } //TODO ? handle incompatible state?
```

Comment:
I was expecting the compiler to have a problem with the fact that this might not return anything?